### PR TITLE
[TensorPipe] Re-enable RPC tests

### PR DESCRIPTION
**Pull request by lw**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #39441 [TensorPipe] Re-enable dist optimizer tests
* #39440 [TensorPipe] Re-enable dist autograd tests
* **#39406 [TensorPipe] Re-enable RPC tests**

For now, just the RPC test (no dist autograd or dist optimizer).

I removed the skipping decorator from all the tests except those that explicitly use the ProcessGroup options.

Includes #39027.

Differential Revision: [D21716069](https://our.internmc.facebook.com/intern/diff/D21716069/)

### CODE REVIEW

File: torch/testing/_internal/distributed/rpc/rpc_test.py [hunk 1]
```diff
@@ -2429,8 +2403,9 @@ def _create_rref(self):
         )
 
     @dist_init
-    @_skip_if_tensorpipe_agent
     def test_user_rrefs_confirmed(self):
+        print(os.getpid())
+        # time.sleep(20)
```
Comment (lw) [inline torch/testing/_internal/distributed/rpc/rpc_test.py:2408]: Oups, these slipped in from my debugging sessions, I'll remove them (and the import above) but I first want to wait for the CI to finish.

File: test/run_test.py [hunk 2]
```diff
@@ -72,6 +72,9 @@
     'distributed/rpc/faulty_agent/test_dist_autograd_spawn',
     'distributed/rpc/faulty_agent/test_rpc_spawn',
     'distributed/rpc/jit/test_dist_autograd_spawn',
+    # 'distributed/rpc/tensorpipe/test_dist_autograd_spawn',
```
Comment (mrshenli) [inline test/run_test.py:75]: Let's remove these commented-out lines from this PR. We can add them back when we are ready.

**Review by mrshenli — APPROVED** [review]
Stamp to unblock, but please wait for tests and remove commented-out lines. Thanks!

File: torch/testing/_internal/distributed/rpc/rpc_test.py [hunk 3]
```diff
@@ -2429,8 +2403,9 @@ def _create_rref(self):
         )
 
     @dist_init
-    @_skip_if_tensorpipe_agent
     def test_user_rrefs_confirmed(self):
+        print(os.getpid())
```
Comment (osalpekar) [inline torch/testing/_internal/distributed/rpc/rpc_test.py:2407]: Are these debugging artifacts that we should remove?

File: torch/testing/_internal/distributed/rpc/rpc_test.py [hunk 4]
```diff
@@ -1,4 +1,5 @@
 import concurrent.futures
+import os
```
Comment (osalpekar) [inline torch/testing/_internal/distributed/rpc/rpc_test.py:2]: Do we need this import? (apart from the os.getpid call below?)

**Review by osalpekar — APPROVED** [review]
Looks good - what do we plan on doing with #39027?

**Pull request closed without merging**